src/entities/swap/types.ts

```diff
@@ -0,0 +1,64 @@
+import { TokenAmount } from '../tokenAmount';
+import { MinimalToken, Slippage } from '../..';
+import { SingleSwap, SwapKind, BatchSwapStep } from '../../types';
+import { PathWithAmount } from './pathWithAmount';
+import { Address, Hex } from 'viem';
+
+export type SwapBuildOutputBase = {
+    to: Address;
+    callData: Hex;
+    value: bigint;
+};
+
+export type SwapBuildOutputExactIn = SwapBuildOutputBase & {
+    minAmountOut: TokenAmount;
+};
+
+export type SwapBuildOutputExactOut = SwapBuildOutputBase & {
+    maxAmountIn: TokenAmount;
+};
+
+export type TokenApi = Omit<MinimalToken, 'index'>;
+
+export type Path = {
+    pools: Address[] | Hex[];
+    tokens: TokenApi[];
+    outputAmountRaw: bigint;
+    inputAmountRaw: bigint;
+    balancerVersion: 2 | 3;
+};
+
+export interface SwapBase {
+    chainId: number;
+    isBatchSwap: boolean;
+    paths: PathWithAmount[];
+    assets: Address[];
+    swapKind: SwapKind;
+    swaps: BatchSwapStep[] | SingleSwap;
+    quote: TokenAmount;
+    inputAmount: TokenAmount;
+    outputAmount: TokenAmount;
+    query(rpcUrl?: string, block?: bigint): Promise<TokenAmount>;
+    queryCallData(): string;
+    buildCall(swapCall: SwapCallBuild): SwapBuildOutputBase;
+}
+
+type BaseSwapCall = {
```

Comment:
A small suggestion for consistency:
```suggestion
type SwapCallBase = {
```